stores/autopilot.go

```diff
@@ -1,117 +1,96 @@
 package stores
 
 import (
+	"context"
 	"encoding/json"
-	"os"
-	"path/filepath"
-	"sync"
-	"time"
+	"errors"
 
 	"go.sia.tech/renterd/api"
-	"go.sia.tech/siad/modules"
+	"gorm.io/gorm"
+	"gorm.io/gorm/clause"
 )
 
-// EphemeralAutopilotStore implements autopilot.Store in memory.
-type EphemeralAutopilotStore struct {
-	mu     sync.Mutex
-	config api.AutopilotConfig
-}
-
-// Config implements autopilot.Store.
-func (s *EphemeralAutopilotStore) Config() api.AutopilotConfig {
-	s.mu.Lock()
-	defer s.mu.Unlock()
-	return s.config
-}
-
-// SetConfig implements autopilot.Store.
-func (s *EphemeralAutopilotStore) SetConfig(c api.AutopilotConfig) error {
-	s.mu.Lock()
-	defer s.mu.Unlock()
-	s.config = c
-	return nil
-}
+type (
+	dbAutopilot struct {
+		Model
 
-// ProcessConsensusChange implements chain.Subscriber.
-func (s *EphemeralAutopilotStore) ProcessConsensusChange(cc modules.ConsensusChange) {
-	panic("not implemented")
-}
+		Identifier    string `gorm:"unique;NOT NULL;"`
+		Config        json.RawMessage
+		CurrentPeriod uint64 `gorm:"index;default:0"`
```

Comment:
Why not avoid the index here and put it on the `Identifier` instead? We are always going to perform lookups by identifier.

Reply:
I believe `unique` adds an index.

I figured an index on `CurrentPeriod` was good practice since we might want to eventually query it for accounting purposes maybe. I'll get rid of it since the cardinality if this table should always be super low anyway.